pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -141,32 +150,44 @@ func (c *Cluster) checkHealth() error {
 	// handle all mons that haven't been in the Ceph mon map
 	for mon := range monsNotFound {
 		logger.Warningf("mon %s NOT found in ceph mon map, failover", mon)
-		c.failMon(len(c.clusterInfo.Monitors), mon)
+		c.failMon(len(c.clusterInfo.Monitors), desiredMonCount, mon)
 		// only deal with one "not found in ceph mon map" mon per health check
 		return nil
 	}
 
-	// check if there are more than two mons running on the same node, failover one mon in that case
-	done, err := c.checkMonsOnSameNode()
-	if done || err != nil {
-		return err
+	if !allowMultiplePerNode {
+		// check if there are more than two mons running on the same node, failover one mon in that case
+		done, err := c.checkMonsOnSameNode(desiredMonCount)
+		if done || err != nil {
+			return err
+		}
 	}
 
-	done, err = c.checkMonsOnValidNodes()
+	done, err := c.checkMonsOnValidNodes()
 	if done || err != nil {
 		return err
 	}
 
-	// create/start new mons when there are less mons
-	if len(status.MonMap.Mons) < c.Size {
-		logger.Infof("found only %d mons less than given mon.count %d, starting more mons", len(status.MonMap.Mons), c.Size)
+	// create/start new mons when there are fewer mons than the desired count in the CRD
+	if len(status.MonMap.Mons) < desiredMonCount {
+		logger.Infof("Adding mons. Currently %d mons are in quorum and the desired count is %d.", len(status.MonMap.Mons), desiredMonCount)
 		return c.startMons()
 	}
 
+	// remove extra mons if the desired count has decreased in the CRD and all the mons are currently healthy
+	if allMonsInQuorum && len(status.MonMap.Mons) > desiredMonCount {
+		if desiredMonCount < 2 && len(status.MonMap.Mons) == 2 {
+			logger.Warningf("cannot reduce mon quorum size from 2 to 1")
+		} else {
+			logger.Infof("Removing an extra mon. Currently %d are in quorum and only %d are desired", len(status.MonMap.Mons), desiredMonCount)
```

Comment:
Please make all log messages begin with a lowercase char, so it is consistent across the code.

Reply:
Is lower case consistent? It doesn't seem consistent with upper case comments. :)